### PR TITLE
chore: [V2] rename 'core' module to 'common'

### PR DESCRIPTION
**Pull request by scottgerring**
**Issue #, if available:**
#1076 

## Description of changes:
The Java 'core' module has nothing to do with the powertools concept of 'core' utilities - logging, tracing, metrics - but is rather common functionality shared by the other PT modules.

To clarify this distinction, we will rename this to 'powertools-common', which fits with the other PT libraries naming scheme.

**Checklist**

 

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [ ] Update tests
* [ ] Update docs
* [ ] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: powertools-common/src/main/java/software/amazon/lambda/powertools/core/internal/LambdaConstants.java
Comment (jeromevdl) [inline powertools-common/src/main/java/software/amazon/lambda/powertools/core/internal/LambdaConstants.java:1]: Package should be changed too

**Comment by scottgerring** [comment]
We still have `examples/powertools-examples-core` . Previously this was super wrong - it had nothing to do with the 'core' module.

Now it is a little less wrong - it is the concept of "core utilities", which are split over `powertools-logging`, `powertools-tracing`, `powertools-metrics`. 

What should we call the example?

**Comment by scottgerring** [comment]
Fixed namespace

File: powertools-common/pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
         <version>2.0.0-SNAPSHOT</version>
     </parent>
 
-    <name>Powertools for AWS Lambda (Java) library Core</name>
+    <name>Powertools for AWS Lambda (Java) library Common Utilities</name>
     <description>
         A suite of utilities for AWS Lambda Functions that makes tracing with AWS X-Ray, structured logging and creating custom metrics asynchronously easier.
```
Comment (jeromevdl) [inline powertools-common/pom.xml:32]: All the content of common is internal, maybe we could simplify this description by just saying common internal utilities...

**Review by jeromevdl — COMMENTED** [review]
couple of comments

File: powertools-common/pom.xml [hunk 2]
```diff
@@ -27,10 +27,8 @@
         <version>2.0.0-SNAPSHOT</version>
     </parent>
 
-    <name>Powertools for AWS Lambda (Java) library Core</name>
-    <description>
-        A suite of utilities for AWS Lambda Functions that makes tracing with AWS X-Ray, structured logging and creating custom metrics asynchronously easier.
-    </description>
+    <name>Powertools for AWS Lambda (Java) library Common Internal Utilities</name>
```
Comment (scottgerring) [inline powertools-common/pom.xml:30]: @jeromevdl how about this (and description)

Comment (jeromevdl) [reply powertools-common/pom.xml:30]: I would change description to something like

> Internal utilities used in Lambda Powertools modules, do not use it directly in your project

Comment (scottgerring) [reply powertools-common/pom.xml:30]: Done

**Review by jeromevdl — APPROVED** [review]
LGTM !

**Pull request merged**